File: LOGICA DE PROGRAMACAO/Java/ws-eclipse/curso-programacao/src/exercicio6/diagonal_negativos.java
package exercicio6;

import java.util.Scanner;

public class diagonal_negativos {
	
	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in);
		
		int N, negativos;
		
		System.out.print("ORDEM DA MATRIZ: ");
		N = sc.nextInt();
		
		int[][] mat = new int[N][N];
		
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				System.out.print("MAT[" + i + "][" + j + "]: ");
				mat[i][j] = sc.nextInt();
				}
		}
		
		System.out.print("DIAGONAL PRINCIPAL DA MATRIZ: ");
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				if (i == j) {
					if (i < N - 1 && j < N - 1) {
						System.out.print(mat[i][j] + ", ");
					} else {
						System.out.print(mat[i][j]);
					}
				}
			}
		}
		
		negativos = 0;
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				if (mat[i][j] < 0) {
					negativos++;
				}
			}
		}
		System.out.println("\nNUMEROS NEGATIVOS PRESENTES NA MATRIZ: " + negativos);
	
		sc.close();
	}

}

File: LOGICA DE PROGRAMACAO/Python/ws-pycharm/exercicios/diagonal_negativos.py
N = int(input("ORDEM DA MATRIZ: "))

mat = [[0 for x in range(N)] for y in range(N)]

negativos = 0
for i in range(0, N):
    for j in range(0, N):
        mat[i][j] = int(input(f"mat[{i},{j}]: "))
        if mat[i][j] < 0:
            negativos += 1

print("\nMATRIZ CRIADA:")
for i in range(0, N):
    for j in range(0, N):
        print(f"|{mat[i][j]}|", end=" ")
    print()

print("\nNumeros na diagonal: ")
for i in range(0, N):
    for j in range(0, N):
        if i == j:
            print(f"|{mat[i][j]}|", end=" ")

print(f"\nNumero de negativos na matriz: {negativos}")